server/src/main/java/org/elasticsearch/search/aggregations/bucket/histogram/AutoDateHistogramAggregationBuilder.java

```diff
@@ -101,12 +128,16 @@ public AutoDateHistogramAggregationBuilder(String name) {
     public AutoDateHistogramAggregationBuilder(StreamInput in) throws IOException {
         super(in, ValuesSourceType.NUMERIC, ValueType.DATE);
         numBuckets = in.readVInt();
+        if (in.getVersion().onOrAfter(Version.V_8_0_0)) {
```

Comment:
Just a procedural note: if we are backporting this to 7.x, the backport will need to change the version, and then we'll need to push a followup commit to master which change the version appropriately.  I like to leave a `#TODO change version after backport` so I don't forget, but that's just personal preference.

Just an FYI since version checks still confuse me from time to time :)

Reply:
believe I addressed this comment